src/java.base/windows/classes/sun/nio/fs/WindowsLinkSupport.java

```diff
@@ -303,10 +303,14 @@ private static String readLinkImpl(long handle) throws IOException {
             try {
                 DeviceIoControlGetReparsePoint(handle, buffer.address(), size);
             } catch (WindowsException x) {
-                // FIXME: exception doesn't have file name
+                String filename = null;
+                try {
+                    filename = GetFinalPathNameByHandle(handle);
```

Comment:
One question is whether the string returned by `GetFinalPathNameByHandle` should be used as is, or converted to some other form.

Reply:
I think you just need to use the path that is passed to the readLink method, just pass it readLinkImpl.